### PR TITLE
[Doc] update iotdb document

### LINKED ISSUE #5399: [Bug] [Docs] IoTDB Source example error
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

source example error

### SeaTunnel Version

dev

### SeaTunnel Config

```conf
-
```

### Running Command

```shell
-
```

### Error Exception

```log
source example error
```

### Zeta or Flink or Spark Version

_No response_

### Java or Scala Version

_No response_

### Screenshots

image

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by liunaijie**
## Purpose of this pull request
1. close #5399 
2. update document format to v2
3. move host,port config as they duplicated with node_url config and node_url is required.

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  4. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by liunaijie** [comment]
@hailin0  PTAL

**Comment by zhilinli123** [comment]
Hi @liunaijie  Would you like to contribute v2.0 documentation to the Iotdb connector? I look forward to your contribution
issues：https://github.com/apache/seatunnel/issues/4572
reference:https://github.com/apache/seatunnel/blob/dev/docs/en/connector-v2/source/Mysql.md
https://github.com/apache/seatunnel/blob/dev/docs/en/connector-v2/sink/Mysql.md

**Comment by liunaijie** [comment]
sure

**Comment by ruanwenjun** [comment]
LGTM, it's better to use v2.0 doc style as @zhilinli123 said.

File: docs/en/connector-v2/sink/IoTDB.md [hunk 1]
```diff
@@ -68,77 +68,46 @@ There is a conflict of thrift version between IoTDB and Spark.Therefore, you nee
...
+        device_name = "string"
+        temperature = "float"
+        moisture = "bigint"
+      }
+    }
   }
 }
-```
 
-Upstream SeaTunnelRow data format is the following:
```
Comment (hailin0) [inline docs/en/connector-v2/sink/IoTDB.md:99]: why delete？

File: docs/en/connector-v2/sink/IoTDB.md [hunk 2]
```diff
@@ -68,77 +68,46 @@ There is a conflict of thrift version between IoTDB and Spark.Therefore, you nee
...
-
-```hocon
 sink {
   IoTDB {
-    ...
+    node_urls = "localhost:6667"
+    username = "root"
+    password = "root"
     key_device = "device_name"
-    key_timestamp = "ts"
```
Comment (hailin0) [inline docs/en/connector-v2/sink/IoTDB.md:129]: why delete？

File: docs/en/connector-v2/sink/IoTDB.md [hunk 3]
```diff
@@ -68,77 +68,46 @@ There is a conflict of thrift version between IoTDB and Spark.Therefore, you nee
...
     key_measurement_fields = ["temperature", "moisture"]
   }
 }
+
 ```
 
 Upstream SeaTunnelRow data format is the following:
 
-|      ts       |       device_name        | field_1 | field_2 | temperature | moisture |
```
Comment (hailin0) [inline docs/en/connector-v2/sink/IoTDB.md:137]: whl update ？

File: docs/en/connector-v2/sink/IoTDB.md (hunk 1 above)
Comment (liunaijie) [reply docs/en/connector-v2/sink/IoTDB.md:99]: the sink example, I only add 1 case (fakesource to iotdb)(also 1 case in source example, use iotdb to console)

the key_timestamp is optional, so i remove it. 
for example data, remove some filed to keep consistent with example code.

**Comment by liunaijie** [comment]
@hailin0 when i update the doc, i find in source, node_url is required, and host, port is optional. but they all represent same thing. 
I see you recommend set host,port in single node, but in this case node_url is still need to fill. is this duplicate? i want update the code, remove host,port option.

**Comment by EricJoy2048** [comment]
We hope that there should be more examples in the document to demonstrate the scenarios in which different parameters are used. In general, we recommend at least three examples.

**Comment by liunaijie** [comment]
> We hope that there should be more examples in the document to demonstrate the scenarios in which different parameters are used. In general, we recommend at least three examples.

updated

File: docs/en/connector-v2/sink/IoTDB.md [hunk 4]
```diff
@@ -68,78 +68,118 @@ There is a conflict of thrift version between IoTDB and Spark.Therefore, you nee
...
+    row.num = 16
+    bigint.template = [1664035200001]
+    schema = {
+      fields {
+        device_name = "string"
+        temperature = "float"
+        moisture = "int"
+        event_ts = "bigint"
+        field_1 = "string"
+        field_2 = "string"
```
Comment (zhilinli123) [inline docs/en/connector-v2/sink/IoTDB.md]: add all type 
image

File: docs/en/connector-v2/source/IoTDB.md [hunk 5]
```diff
@@ -157,37 +106,31 @@ Source plugin common parameters, please refer to [Source Common Options](common-
...
     node_urls = "localhost:6667"
     username = "root"
     password = "root"
+    sql = "SELECT temperature, moisture FROM root.test_group.* WHERE time < 4102329600000 align by device"
+    schema {
+      fields {
+        ts = timestamp
+        device_name = string
+        temperature = float
+        moisture = bigint
```
Comment (zhilinli123) [inline docs/en/connector-v2/source/IoTDB.md:126]: image
ditto

File: docs/en/connector-v2/source/IoTDB.md [hunk 6]
```diff
@@ -18,106 +20,53 @@ supports query SQL and can achieve projection effect.
...
+| INT32           | SMALLINT            |
+| INT32           | INT                 |
+| INT64           | BIGINT              |
+| FLOAT           | FLOAT               |
+| DOUBLE          | DOUBLE              |
+| TEXT            | STRING              |
+
+## Source Options
+
+|            name            |  type   | required | default value |                                    Description                                     |
```
Comment (zhilinli123) [inline docs/en/connector-v2/source/IoTDB.md]: |            name            |  type   | required | default value |                                    Description                                     |
|            Name            |  Type   | Required | Default Value |                                  Description                                     |

File: docs/en/connector-v2/sink/IoTDB.md [hunk 7]
```diff
@@ -68,78 +68,124 @@ There is a conflict of thrift version between IoTDB and Spark.Therefore, you nee
...
 
 ```hocon
 sink {
   IoTDB {
-    node_urls = ["localhost:6667"]
+    node_urls = "localhost:6667"
     username = "root"
     password = "root"
-    batch_size = 1024
+    key_device = "device_name"
```
Comment (zhilinli123) [inline docs/en/connector-v2/sink/IoTDB.md]: Add some description to this? What does the configuration need to pay attention to to tell us the user what it does?

File: docs/en/connector-v2/sink/IoTDB.md [hunk 8]
```diff
@@ -68,78 +68,124 @@ There is a conflict of thrift version between IoTDB and Spark.Therefore, you nee
...
 
 ```hocon
 sink {
   IoTDB {
-    ...
+    node_urls = "localhost:6667"
+    username = "root"
+    password = "root"
     key_device = "device_name"
+    key_timestamp = "event_ts"
```
Comment (zhilinli123) [inline docs/en/connector-v2/sink/IoTDB.md]: Add some description to this? What does the configuration need to pay attention to to tell us the user what it does?

**Review by ruanwenjun — APPROVED** [review]
LGTM

**Review by zhilinli123 — APPROVED** [review]

**Comment by zhilinli123** [comment]
thanks contribution PTAL: @EricJoy2048

**Review by liugddx — APPROVED** [review]

**Pull request merged**